Lesson_05/homework_05_1.py

```diff
@@ -0,0 +1,98 @@
+            'Porsche': ('black', 2017, 3.0, 'coupe', 80000),
+            'Jeep': ('green', 2021, 3.0, 'suv', 50000),
+            'Chrysler': ('gray', 2016, 2.4, 'sedan', 22000),
+            'Dodge': ('yellow', 2020, 3.6, 'suv', 40000),
+            'Ferrari': ('red', 2019, 4.0, 'coupe', 500000),
+            'Lamborghini': ('orange', 2021, 5.0, 'coupe', 800000),
+            'Maserati': ('blue', 2018, 4.7, 'coupe', 100000),
+            'Bugatti': ('black', 2020, 8.0, 'coupe', 2000000),
+            'McLaren': ('yellow', 2017, 4.0, 'coupe', 700000),
+            'Rolls-Royce': ('white', 2019, 6.8, 'sedan', 500000),
+            'Bentley': ('gray', 2020, 4.0, 'coupe', 300000),
+            'Jaguar': ('red', 2016, 2.0, 'suv', 40000),
+            'Land Rover': ('green', 2018, 3.0, 'suv', 60000),
+            'Tesla': ('silver', 2020, 0.0, 'sedan', 60000),
+            'Acura': ('white', 2017, 2.4, 'suv', 40000),
+            'Cadillac': ('black', 2019, 3.6, 'suv', 55000),
+            'Infiniti': ('gray', 2018, 2.0, 'sedan', 35000),
+            'Lincoln': ('white', 2021, 2.0, 'suv', 50000),
+            'GMC': ('blue', 2016, 1.5, 'pickup', 30000),
+            'Ram': ('black', 2019, 5.7, 'pickup', 40000),
+            'Chevy': ('red', 2017, 2.4, 'pickup', 35000),
+            'Dodge Ram': ('white', 2020, 3.6, 'pickup', 45000),
+            'Ford F-Series': ('gray', 2021, 3.5, 'pickup', 50000),
+            'Nissan Titan': ('silver', 2018, 5.6, 'pickup', 35000),
+            }
+
+# We have search_criteria as tuple of ( year>= , engine_volume >= , price<=)
+search_criteria = (2017, 1.6, 36000)
+
+
+# Option 01.
+# write code that will help us to get cars that satisfy search_criteria.
+# Create new dictionary filtrated by search_criteria value condition.
+filtered_cars = {key: value for key, value in car_data.items()
+                 if value[1] >= search_criteria[0]
+                 and value[2] >= search_criteria[1]
+                 and value[4] <= search_criteria[2]}
+
+# Sort by price ascending.
+# We should print up to five (5) first found elements
+cars = sorted(filtered_cars.items(), key=lambda value: value[1][4])
+
+# display up to five (5) first found elements.
+option1 = f'OPTION # 1: {dict(cars[:5])}'
+_log.info(option1)
+
+
+# Option 02.
+# Create new list and sort by price ascending.
+filtered_cars_list = sorted(car_data.items(), key=lambda value: value[1][4])
+
+# Filter created list by search_criteria values.
+filtered_cars_list = [value for value in filtered_cars_list
+                      if value[1][1] >= search_criteria[0]
+                      and value[1][2] >= search_criteria[1]
+                      and value[1][4] <= search_criteria[2]]
+
+# Display up to five (5) first found elements.
+option2 = f'OPTION # 2: {dict(filtered_cars_list[:5])}'
+_log.info(option2)
+
+# Compare values from Option 1 and Option 2.
+_log.info(dict(cars[:5]) == dict(filtered_cars_list[:5]))
```

Comment:
Try to avoid hardcodes in your code - put all hardcoded values as constants at the top of the module